streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -519,6 +563,33 @@ private void handleTasksPendingInitialization() {
         }
     }
 
+    private void handleInitialTaskReuse(final Map<TaskId, Set<TopicPartition>> activeTasksToCreate,
+                                        final Map<TaskId, Set<TopicPartition>> standbyTasksToCreate,
+                                        final Map<TaskId, RuntimeException> failedTasks) {
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToRecycle = assignInitialTasks(activeTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToUse = assignInitialTasks(standbyTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+
+        // if this was the last local thread to receive its assignment, close all the remaining Tasks, as they are not needed
+        stateDirectory.closeInitialTasksIfLastAssginedThread();
+
+        // recycle the initial standbys to active, and remove them from the set of actives that need to be created
+        if (!initialStandbyTasksToRecycle.isEmpty()) {
+            final Set<Task> tasksToCloseDirty = new HashSet<>();
+            for (final Map.Entry<Task, Set<TopicPartition>> entry : initialStandbyTasksToRecycle.entrySet()) {
+                final Task task = entry.getKey();
+                final Set<TopicPartition> inputPartitions = entry.getValue();
+                recycleTaskFromStateUpdater(task, inputPartitions, tasksToCloseDirty, failedTasks);
+                activeTasksToCreate.remove(task.id());
+            }
+        }
+
+        // use initial Standbys as real Standby tasks
+        if (!initialStandbyTasksToUse.isEmpty()) {
```

Comment:
Same as above? If empty, nothing will happen anyway as `keySet()` will be empty?